### DISCUSSION TITLE
New-PnPPersonalSite

### DISCUSSION THREAD

**Discussion by McFarTech** [pnp/powershell#1715] [General]
Guys. 

Can anyone tell me what is wrong with this code please. I am trying to obtain the UPN name and then pass the UPN to the New-PnPPersonalSite -Email command to create the users OneDrive. 

I get the error message

Attempted to perform an unauthorized operation. We dont use legacy protocols so cant use clientsecret. 

I am just not sure why else to do here. 

$connectionName = "AzureRunAsConnection"
$servicePrincipalConnection = Get-AutomationConnection –Name $connectionName  
$TenantName = "TenanatName.onmicrosoft.com"
$AdminSiteURL="https://AdminSiteName-admin.sharepoint.com"

Connect-PnPOnline -ClientId $servicePrincipalConnection.ApplicationID -Url $AdminSiteURL -Tenant $TenantName -Thumbprint $servicePrincipalConnection.CertificateThumbprint

$OneDriveUser = Get-PnPAzureADUser | Where-Object {$_.officelocation -eq "ABC"} | select-object UserPrincipalName | sort-object UserPrincipalName

Foreach($UPN in $OneDriveUser)
{
    Write-output "Adding Site Collection Admin to:" $UPN.UserPrincipalName
    New-PnPPersonalSite -Email $UPN.UserPrincipalName
}

**Comment by McFarTech** [comment]
This is the error i see when running the script above. 

UserPrincipalName               
-----------------               
firstname.lastname@domainname.com
Attempted to perform an unauthorized operation.
Adding Site Collection Admin to:

**Comment by ztrhgf** [comment]
Same behavior when `Connect-PnPOnline -Url $url -Interactive -ForceAuthentication` was used for authentication. If `Connect-PnPOnline -Url $url -UseWebLogin -ForceAuthentication` is used, error "Attempted to perform an unauthorized operation" is **not** thrown.
